File: PL/code_practice/PL6/list-ave.c
#include <stdio.h>

double list_ave(int *list, int size) {
    if (size == 0) {
        return 0.0;
    }
    int sum = 0;
    for (int i = 0; i < size; i++) {
        sum += list[i];
    }
    return (double)sum / size;
}

int main() {
    int arr[] = {1, 2, 3, 4, 5};
    int size = sizeof(arr) / sizeof(arr[0]);
    double average = list_ave(arr, size);
    printf("리스트의 평균: %.2f\n", average);
    return 0;
}

File: PL/code_practice/PL6/list-ave.py
def list_ave(lst):
    if not lst:
        return 0
    return sum(lst) / len(lst)

if __name__ == "__main__":
    line = input("정수 리스트 입력 (공백으로 구분): ").strip()
    if not line:
        print("입력값이 비어 있습니다. 하나 이상의 정수를 입력하세요.")
        raise SystemExit(1)
    lst = list(map(int, line.split()))
    ave = list_ave(lst)
    print(f"리스트의 평균: {ave:.2f}")
